src/System.Buffers.Primitives/System/Buffers/OwnedBuffer.cs

```diff
@@ -69,7 +40,11 @@ public void Dispose()
         public abstract void Retain();
 
         public abstract void Release();
-
         #endregion
+
+        protected static unsafe void* Add(void* pointer, int offset)
```

Comment:
What about using [Unsafe.Add](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.CompilerServices.Unsafe/ref/System.Runtime.CompilerServices.Unsafe.cs#L13)? We use it in Primitive Formatters a lot.